test/e2e/reconciler/cache/replication_test.go

```diff
@@ -60,109 +61,264 @@ type testScenario struct {
 // scenarios all test scenarios that will be run against in-process and standalone cache server
 var scenarios = []testScenario{
 	{"TestReplicateAPIExport", replicateAPIExportScenario},
+	{"TestReplicateAPIResourceSchema", replicateAPIResourceSchemaScenario},
+	{"TestReplicateAPIExportNegative", replicateAPIExportNegativeScenario},
+}
+
+// baseScenario an auxiliary struct that is used by replicateResourceScenario
+type baseScenario struct {
+	server framework.RunningServer
+
+	resourceName string
+	resourceKind string
+
+	createSourceResource        func(logicalcluster.Name) error
+	getSourceResource           func(logicalcluster.Name) (interface{}, error)
+	updateSourceResource        func(logicalcluster.Name, interface{}) (interface{}, error)
+	updateSpecForSourceResource func(interface{}) error
+	deleteSourceResource        func(logicalcluster.Name) error
+
+	getCachedResource func(logicalcluster.Name) (interface{}, error)
+}
+
+// replicateAPIResourceSchemaScenario tests if an APIResourceSchema is propagated to the cache server.
+// The test exercises creation, modification and removal of the APIExport object.
+func replicateAPIResourceSchemaScenario(ctx context.Context, t *testing.T, server framework.RunningServer, kcpShardClusterClient clientset.ClusterInterface, cacheKcpClusterClient clientset.ClusterInterface) {
+	replicateResourceScenario(t, baseScenario{
+		server:       server,
+		resourceName: "today.sheriffs.wild.wild.west",
+		resourceKind: "APIResourceSchema",
+		createSourceResource: func(cluster logicalcluster.Name) error {
+			apifixtures.CreateSheriffsSchemaAndExport(ctx, t, cluster, kcpShardClusterClient.Cluster(cluster), "wild.wild.west", "testing replication to the cache server")
+			return nil
+		},
+		getSourceResource: func(cluster logicalcluster.Name) (interface{}, error) {
+			return kcpShardClusterClient.Cluster(cluster).ApisV1alpha1().APIResourceSchemas().Get(ctx, "today.sheriffs.wild.wild.west", metav1.GetOptions{})
+		},
+		updateSourceResource: func(cluster logicalcluster.Name, res interface{}) (interface{}, error) {
+			apiResSchema, ok := res.(*apisv1alpha1.APIResourceSchema)
+			if !ok {
+				return nil, fmt.Errorf("%T is not *APIResourceSchema", res)
```

Comment:
With generics you don't need to write this code

Reply:
yeah, this is the only case that would benefit from using generics, I can look into that in a follow-up PR.